src/vision/frc_vision_final.py

```diff
@@ -118,112 +129,138 @@ def newLine(filtered_lines, new_line, filtered_line_img, x1, y1, x2, y2, X_TOTAL
     if isShooting:
         # Convert from RGB to HSV, helps with filtering
         HSV = cv2.cvtColor(IMG, cv2.COLOR_BGR2HSV)
-    
+
         # Define upper and lower bounds for HSV variables
         LOWER_COLOR = np.array([70, 80, 255])
         UPPER_COLOR = np.array([95, 180, 255])
-    
+
         # Create mask within hsv range
         MASK = cv2.inRange(HSV, LOWER_COLOR, UPPER_COLOR)
-    
-        # Various blur method testings (buffer for pixel imperfections)
-        # BLUR = cv2.GaussianBlur(MASK, (3, 3), 0)
+
+        # Blur image (buffer for pixel imperfections)
         MEDIAN = cv2.medianBlur(MASK, 5)
-    
-        # Edge detection on each test for use in line detection
-        # BLUR_EDGES = cv2.Canny(BLUR, 100, 200)
-        # MASK_EDGES = cv2.Canny(MASK, 100, 200)
+
+        # Edge detection for use in line detection
         MED_EDGES = cv2.Canny(MEDIAN, 50, 150)
-    
-        # Empty image for drawing lines (testing)
-        FILTERED_LINE_IMG = np.zeros((HEIGHT, WIDTH, 3), np.uint8)
-        LINE_IMG = np.zeros((HEIGHT, WIDTH, 3), np.uint8)
-    
+
+        if debugging:
+            BLUR = cv2.GaussianBlur(MASK, (3, 3), 0)
+
+            BLUR_EDGES = cv2.Canny(BLUR, 100, 200)
+            MASK_EDGES = cv2.Canny(MASK, 100, 200)
+
+            # Empty image for drawing lines (testing)
+            FILTERED_LINE_IMG = np.zeros((HEIGHT, WIDTH, 3), np.uint8)
+            LINE_IMG = np.zeros((HEIGHT, WIDTH, 3), np.uint8)
+
         # Find lines in selected image
-        LINES = cv2.HoughLinesP(MED_EDGES, 1, radians(.5), 25, maxLineGap=25)
-    
+        LINES = cv2.HoughLinesP(MED_EDGES, 1, radians(0.5), 25, maxLineGap=25)
+
         # If there are lines:
         if LINES is not None:
             NUM_LINES = len(LINES)
             FILTERED_LINES = []
             X_TOTAL = 0
             Y_TOTAL = 0
-    
+
             for NEW_LINE in LINES:
                 x1, y1, x2, y2 = NEW_LINE[0]
-                new_slope = degrees(np.arctan((y2 - y1)/(x2 - x1)))
-    
+                new_slope = degrees(np.arctan((y2 - y1) / (x2 - x1)))
+
                 # Checks if we have verified lines, and makes a new line based on that.
                 if FILTERED_LINES:
-                    if (new_slope < -ANGLE_THRESHOLD or new_slope > ANGLE_THRESHOLD) and unequal(new_slope, FILTERED_LINES):
+                    if (
+                        new_slope < -ANGLE_THRESHOLD or new_slope > ANGLE_THRESHOLD
```

Comment:
Shouldn't we have two different ANGLE_THRESHOLD values for the two sides of the target?